### PR TITLE
Fix buffer

### PR DESCRIPTION
**Pull request by marc9595**
Fix https://github.com/tinyclues/fletcher/issues/29.

`np.frombuffer` was used in a wrong way because the value of `offset` has to be `chunk.offset * sizeof(chunk.type)`. I rewrite the code so that this won't be an issue.

Also, `max` and `min` didn't handle the case where the null buffer is None so I fixed that as well.

### CODE REVIEW

File: fletcher/_algorithms.py [hunk 1]
```diff
@@ -269,13 +269,15 @@ def aggregate_fletcher_array(fr_arr, aggregator):
...
-            count=len(chunk),
-            offset=chunk.offset,
-        )
-        op_ma = {"max": max_ma, "min": min_ma}[aggregator]
-        return op_ma(arr_buff, chunk.buffers()[0])
+            chunk.buffers()[1], dtype=chunk.type.to_pandas_dtype()
+        )[chunk.offset : chunk.offset + len(chunk)]
+        null_buff = chunk.buffers()[0]
+        if null_buff is None:
+            op_ma = {"max": np.amax, "min": np.amin}[aggregator]
```
Comment (krivonogov) [inline fletcher/_algorithms.py]:
```suggestion
            op_ma = {"max": np.max, "min": np.min}[aggregator]
```

File: fletcher/_algorithms.py [hunk 2]
```diff
@@ -269,13 +269,15 @@ def aggregate_fletcher_array(fr_arr, aggregator):
...
-            dtype=chunk.type.to_pandas_dtype(),
-            count=len(chunk),
-            offset=chunk.offset,
-        )
-        op_ma = {"max": max_ma, "min": min_ma}[aggregator]
-        return op_ma(arr_buff, chunk.buffers()[0])
+            chunk.buffers()[1], dtype=chunk.type.to_pandas_dtype()
+        )[chunk.offset : chunk.offset + len(chunk)]
+        null_buff = chunk.buffers()[0]
+        if null_buff is None:
```
Comment (artemru) [inline fletcher/_algorithms.py]: btw, `chunk.null_count == 0` maybe more explicit ?..

**Comment by krivonogov** [comment]
let's merge this one?

**Comment by marc9595** [comment]
Merging.

**Pull request merged**